lenet5/model.py

```diff
@@ -0,0 +1,40 @@
+"""
+Small 5-layer CNN workload
+03/16/2020 Bandhav Veluri
+"""
+
+import torch
+import torch.nn as nn
+
+class LeNet5(nn.Module):
+    """
+    LeNet-5
+
+    https://cs.nyu.edu/~yann/2010f-G22-2565-001/diglib/lecun-98.pdf
+    (Page 7)
+    """
+    def __init__(self):
+        super(LeNet5, self).__init__()
+
+        self.conv = nn.Sequential(
+            nn.Conv2d(1, 6, kernel_size=(5, 5)),
+            nn.ReLU(),
+            nn.MaxPool2d(kernel_size=(2, 2), stride=2),
+            nn.Conv2d(6, 16, kernel_size=(5, 5)),
+            nn.ReLU(),
+            nn.MaxPool2d(kernel_size=(2, 2), stride=2),
+            nn.Conv2d(16, 120, kernel_size=(5, 5)),
+        )
+
```

Comment:
I think it is possible to merge these two Sequential blocks by having defining a layer that does the `view`

Reply:
There's been lot of debate on this in pytorch forums ;). PyTorch officially recommended way is to not have separate `View` layer.

https://discuss.pytorch.org/t/non-legacy-view-module/131
https://discuss.pytorch.org/t/what-is-reshape-layer-in-pytorch/1110

Reply:
I guess the class constructor should define the network architecture and `forward` defines the dataflow in the network. `View` is just a dataflow tweak and cannot be considered as part of the network architecture...

Reply:
I think this is a good compromise. So our policy is the constructor defines the network architecture ... and forward can reshape inputs if necessary ...